File: ProductiviMe/src/main/java/com/eitan/productivime/BangForYourBuck/ScheduleResult.java
package com.eitan.productivime.BangForYourBuck;

import java.util.List;

// JSON response shape for POST /schedule/generate.
// Spring Boot / Jackson serialises this automatically.
public record ScheduleResult(
        List<ScheduledInterval> intervals,
        int totalValue
) {
    public record ScheduledInterval(
            String start,
            String end,
            int durationMinutes,
            List<ScheduledActivity> activities
    ) {}

    public record ScheduledActivity(
            String name,
            int durationMinutes,
            int value
    ) {}
}

File: ProductiviMe iOS/ProductiviMe iOS/Engine/ScheduleResult.swift
import Foundation

struct ScheduleResult {
    let intervals: [ScheduledInterval]
    let totalValue: Int   // sum of scoringValue (1–10) of scheduled activities

    struct ScheduledInterval: Identifiable {
        var id: String { "\(start)-\(end)" }
        let start: String
        let end: String
        let durationMinutes: Int
        let activities: [ScheduledActivity]
    }

    struct ScheduledActivity: Identifiable {
        var id: String { name }
        let name: String
        let durationMinutes: Int
        let value: Int  // scoringValue (1–10 or 12), not the 3000 sentinel
    }
}
